lsp.go

```diff
@@ -346,26 +363,49 @@ func checkFile(fname string, content string) error {
 		if strings.HasSuffix(filename, ".tm") || strings.HasSuffix(filename, ".tm.hcl") {
 			path := filepath.Join(dir, filename)
 
-			if path == fname {
-				// file already added
+			if path == fromFile {
+				// ignore source file
 				continue
 			}
 
-			contents, err := os.ReadFile(path)
-			if err != nil {
-				log.Error().Err(err).Send()
-				return err
-			}
+			files = append(files, path)
+		}
+	}
 
-			err = parser.AddFile(path, contents)
-			if err != nil {
-				log.Error().Err(err).Send()
-				return err
-			}
+	return files, nil
+}
+
+// checkFiles checks if the given provided files have errors but the thisFile
+// is handled separately because it can be unsaved.
+func checkFiles(files []string, thisFile string, thisContent string) error {
+	dir := filepath.Dir(thisFile)
+	parser := hcl.NewTerramateParser(dir)
+
+	for _, fname := range files {
+		var (
+			contents []byte
+			err      error
+		)
+
+		if thisFile == fname {
+			contents = []byte(thisContent)
+		} else {
+			contents, err = os.ReadFile(fname)
+		}
+
+		if err != nil {
+			log.Error().Err(err).Send()
```

Comment:
Maybe log the errors on the top of the call chain to avoid logging errors twice ? This function is called by another function before sending the actual error, so maybe log there ? (like logging on the last moment possible before sending the info).

Reply:
I think it makes sense, I will have a look in the actual logs. Thanks